shinydashboard/lantern/app.R

```diff
@@ -0,0 +1,231 @@
+#
+# This is a Shiny web application. You can run the application by clicking
+# the 'Run App' button above.
+#
+# Find out more about building applications with Shiny here:
+#
+#    http://shiny.rstudio.com/
+#
+
+library(shiny)
+library(shinydashboard)
+library(DBI)
+library(readr)
+library(dplyr)
+library(RPostgres)
+library(ggplot2)
+library(plotly)
+library(scales)
+library(forcats)
+library(dygraphs)
+library(config)
+library(here)
+
+http_response_code_tbl <- read_csv("./http_codes.csv") %>% mutate(code_chr=as.character(code))
+
+db_config <- config::get("lantern")
+
+con <- dbConnect(RPostgres::Postgres(),
+                 dbname = db_config$database, 
+                 host = db_config$server, # i.e. 'ec2-54-83-201-96.compute-1.amazonaws.com'
+                 port = db_config$port, 
+                 user = db_config$uid,
+                 password = db_config$pwd
+)
+
+fhir_endpoints  <- tbl(con, "fhir_endpoints")
+metrics_values  <- tbl(con, "metrics_values")
+metrics_labels  <- tbl(con, "metrics_labels")
+end_org         <- tbl(con, "endpoint_organization")
+hit_prod        <- tbl(con, "healthit_products")
+endpoint_export <- tbl(con, "endpoint_export")
+
+fhir_endpoints_tbl <- as_tibble(fhir_endpoints)
+
+http_response_ids <- metrics_labels %>%
+    filter(metric_name == "AllEndpoints_http_request_responses") %>%
+    select(id)
+
+https_response_values <- metrics_values %>%
+    right_join(http_response_ids, by = c("labels_id" = "id"))
+
+curr_http_response_tally <- fhir_endpoints_tbl %>%
+    select(http_response) %>%
+    group_by(http_response) %>%
+    tally()
+
+
+totals <- list()
+totals$all_endpoints <- nrow(fhir_endpoints_tbl)
+totals$indexed_endpoints <- nrow(fhir_endpoints_tbl %>% filter(http_response != 0))
+
+# Get the list of most recent HTTP responses when requesting the capability statement from the 
+# fhir_endpoints
```

Comment:
It would be nice to have more comments like this for the various definitions before the dashboard page is defined. Most of them seem pretty self explanatory, but there's a few like http_pctf and foo that aren't clear (at least to me) on first glance!

Reply:
Will restructure the various data processing parts and include comments for each.